maven/detect.go

```diff
@@ -33,8 +36,19 @@ const (
 type Detect struct{}
 
 func (Detect) Detect(context libcnb.DetectContext) (libcnb.DetectResult, error) {
-	file := filepath.Join(context.Application.Path, "pom.xml")
-	_, err := os.Stat(file)
+	l := bard.NewLogger(ioutil.Discard)
+	cr, err := libpak.NewConfigurationResolver(context.Buildpack, &l)
+	if err != nil {
+		return libcnb.DetectResult{}, err
+	}
+
+	pomFile, _ := cr.Resolve("BP_MAVEN_POM_FILE")
+	if pomFile == "" {
```

Comment:
I don't believe that you need this `if` block. The ConfigurationResolver should pull back the value in the env variable or the default value that you set in buildpack.toml.

You do have to be careful for tests though as tests don't automatically read buildpack.toml, so you have to add the configuration in your test setup.

Reply:
I did this to make the tests pass, I'll move it to the test code.